tools/imas/imas_wrapper.py

```diff
@@ -0,0 +1,542 @@
+
+	# load the coreprof psi
+	if len(coreprof.profiles_1d[0].grid.psi) == 0:
+		logging.warning('The poloidal flux from core profiles IDS seems to be empty!\n')
+		raise Exception('Error in loading core profiles data. Please check the input data!')
+	else:
+		cp_psi = coreprof.profiles_1d[0].grid.psi
+		if log:
+			logging.info('The poloidal flux from core profiles IDS loaded successfully!\n')
+	
+	# load the coreprof radius
+	single_cp_radius = coreprof.profiles_1d[0].grid.rho_tor
+	
+	# interpolate the coreprof rho from the coreprof psi to the equilibrium psi
+	interp_radius = CubicSpline(-cp_psi, single_cp_radius)
+	radius = interp_radius(-eq_psi)
+	
+	# Check if the interpolation gave a correct answer
+	error = (radius[-1] - single_cp_radius[-1]) / radius[-1]
+	if abs(error) > 0.1:
+		raise Exception("The interpolation gave significant difference in the equilibrium and coreprofile radii. Cannot continue, please check the input parameters.")
+	
+	# load wall data with error handling
+	if wall.description_2d[0].vessel.unit[0].annular.resistivity == -9e+40:
+		logging.warning('The the wall resistivity from wall IDS seems to be empty!\n')
+		raise Exception('Error in loading wall data. Please check the input data!')
+	else:
+		resistivity = wall.description_2d[0].vessel.unit[0].annular.resistivity
+		if log:
+			logging.info('The major radius of the magnetic axis from core profiles IDS loaded successfully!\n')
+
+
+	if setUpDream:
+		
+		ds = DREAMSettings()
+		
+		ds.radialgrid.setCustomGridPoints(radius)
+
+		if wall_radius == -999:
+			raise Exception('If you want to set up a dream settings object, please specify the wall radius!')
+		else:
+			ds.radialgrid.setWallRadius(wall_radius)
+			
+		ds.radialgrid.setMajorRadius(R)
+		ds.radialgrid.setB0(abs(B0[0]))
+		
+		cp_time_length = len(cp_time)
+		
+		for i in range(len(ion_names)):
+				
+			ds.eqsys.n_i.addIon(ion_names[i], Z=ion_charges[i], iontype=Ions.IONS_PRESCRIBED, Z0=ion_charges[i], n=ion_densities[i*cp_time_length:i*cp_time_length + cp_time_length], r=cp_radius[:,0], t=cp_time)
+			
+		ds.eqsys.T_cold.setPrescribedData(cp_Te, radius=cp_radius[:,0], times=cp_time)
+		ds.eqsys.T_cold.setType(Tcold.TYPE_PRESCRIBED)
+		
+		if e_field!=-999:
+			ds.eqsys.E_field.setType(ElectricField.TYPE_PRESCRIBED)
+			ds.eqsys.E_field.setPrescribedData(efield=cp_efield, radius=cp_radius[:,0], times=cp_time)
+		
+		else:
+			inverse_wall_time = resistivity / (4*np.pi*1e-7*R*np.log(R/wall_radius))
+			if log:
+				logging.info('The iverse wall time is %s 1/s with the given wall radius.\n', inverse_wall_time)
+		
+			ds.eqsys.E_field.setType(ElectricField.TYPE_SELFCONSISTENT)
+			ds.eqsys.E_field.setBoundaryCondition(ElectricField.BC_TYPE_SELFCONSISTENT, inverse_wall_time=inverse_wall_time, R0=R)
+		
+		ds.eqsys.j_ohm.setInitialProfile(eq_jtor[:,0], radius=eq_radius[:,0], Ip0=eq_ip[0])
+		
+		return ds
+	
+	else:
+		print('return a list? hdf5?')
+		return 'TO DO'
+	
+	dataEntry.close()
```

Comment:
This will never be executed, right? You always return before this point?

Reply:
You are right, I added this to line 500 (in the newly edited file, see it after I uploaded everything), bfore the if-else part at the end of the function.